### PR TITLE
Add XML path expression support

### PR DESCRIPTION
**Pull request by johnsonr**
Enables XML editing using path expressions.

Also upgrades libraries, including `tree-path` and `antlr`

### CODE REVIEW

File: test/xml/xmldocFileParserTest.ts [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+        it("should parse POM", async () => {
+            const f = new InMemoryProjectFile("pom.xml", springBootPom());
+            const parsed = await new XmldocFileParser().toAst(f);
+            // visit(parsed, positionVerifier(RootAndSingleChildElement));
+            assert(!!parsed);
+        });
+
+    });
+
+    describe("with path expressions", () => {
```
Comment (lievendoclo) [inline test/xml/xmldocFileParserTest.ts]: Can you add an example where you add an author underneath the list of current authors (so the authors are still grouped together?)

File: tslint.json [hunk 2]
```diff
@@ -15,6 +15,7 @@
             true,
             "never-prefix"
         ],
+        "max-classes-per-file": [],
```
Comment (lievendoclo) [inline tslint.json]: https://github.com/atomist/sdm/blob/master/tslint.json sets this to 7. Do we want to keep that consistent?

Comment (johnsonr) [reply tslint.json]: I think it was defaulting to 1 here and I just quickly added something to fix it.

Yes, we do want to keep it consistent

File: test/xml/xmldocFileParserTest.ts (hunk 1 above)
Comment (johnsonr) [reply test/xml/xmldocFileParserTest.ts]: I've added a simple example of this. It really needs support for the `last()` functionality from Path. I'll raise an issue for that in `tree-path`

File: tslint.json (hunk 2 above)
Comment (johnsonr) [reply tslint.json]: Done

**Comment by johnsonr** [comment]
On it. Not easy

**Comment by johnsonr** [comment]
Going to redo this on `master` with just XML stuff, as rebasing isn't working out

**Pull request closed without merging**